frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -208,6 +209,31 @@ local footerTextGeneratorMap = {
             return ("%d / %d"):format(footer.position, footer.doc_height)
         end
     end,
+    pages_left_book = function(footer)
+        if footer.pageno then
+            if footer.ui.pagemap and footer.ui.pagemap:wantsPageLabels() then
+                -- (Page labels might not be numbers)
+                return ("%s / %s"):format(footer.ui.pagemap:getCurrentPageLabel(true),
+                                          footer.ui.pagemap:getLastPageLabel(true))
+            end
+            if footer.ui.document:hasHiddenFlows() then
+                -- i.e., if we are hiding non-linear fragments and there's anything to hide,
+                local flow = footer.ui.document:getPageFlow(footer.pageno)
+                local page = footer.ui.document:getPageNumberInFlow(footer.pageno)
+                local pages = footer.ui.document:getTotalPagesInFlow(flow)
+                if flow == 0 then
+                    return ("%d // %d"):format(page, pages)
+                else
+                    return ("[%d / %d]%d"):format(page, pages, flow)
+                end
+            else
+                local remaining = footer.pages - footer.pageno
+                return ("-%d / %d"):format(remaining, footer.pages)
```

Comment:
Looks like a quick copy and paste, where you just updated with `-%d` the most used branch. But the others (pagemap, hiddent flow, footer.position) might need to be updated too for this option to be consistent.

Reply:
It is, yes. I can try revising the other branches but I don't know how to test them :)

Reply:
Well, for footer.position, not sure when/if it is used, but it should logically be:
`return ("%d / %d"):format(footer.doc_height - footer.position, footer.doc_height)`

For pagemap, well, these are strings, so you can't do any arithmetic with them. I guess you can let them as-is then.
For hidden flows, you need an EPUB with non-linear stuff. Dunno where to find one. But it should be similar to what you did:
```diff
                local page = footer.ui.document:getPageNumberInFlow(footer.pageno)
                local pages = footer.ui.document:getTotalPagesInFlow(flow)
+               local remaining = pages - page
+               -- and use it in the return -%d
```